src/main/java/com/minecolonies/coremod/entity/ai/citizen/beekeeper/EntityAIWorkBeekeeper.java

```diff
@@ -0,0 +1,460 @@
+
+public class EntityAIWorkBeekeeper extends AbstractEntityAIInteract<JobBeekeeper>
+{
+    /**
+     * Amount of animals needed to bread.
+     */
+    private static final int NUM_OF_ANIMALS_TO_BREED = 2;
+    private static final int BEES_PER_LEVEL          = 3;
+
+    /**
+     * Experience given per beehive harvested.
+     */
+    private static final double EXP_PER_HARVEST = 5.0;
+
+    /**
+     * Distance two animals need to be inside to breed.
+     */
+    private static final int DISTANCE_TO_BREED = 10;
+
+    /**
+     * Distance around a hive wher bees wander.
+     */
+    private static final int HIVE_BEE_RADIUS = 22;
+
+    /**
+     * Delays used to setDelay()
+     */
+    private static final int DECIDING_DELAY   = 40;
+    private static final int NO_ANIMALS_DELAY = 100;
+    private static final int BREEDING_DELAY   = 40;
+
+    /**
+     * Creates the abstract part of the AI. Always use this constructor!
+     *
+     * @param job the job to fulfill
+     */
+    public EntityAIWorkBeekeeper(@NotNull JobBeekeeper job)
+    {
+        super(job);
+        super.registerTargets(
+          new AITarget(IDLE, START_WORKING, 1),
+          new AITarget(START_WORKING, this::startWorkingAtOwnBuilding, TICKS_SECOND),
+          new AITarget(PREPARING, this::prepareForHerding, 1),
+          new AITarget(DECIDE, this::decideWhatToDo, 1),
+          new AITarget(HERDER_BREED, this::breedAnimals, 1),
+          new AITarget(BEEKEEPER_HARVEST, this::harvestHoney, TICKS_SECOND)
+        );
+    }
+
+    public boolean maxAnimals(final List<BeeEntity> allAnimals)
+    {
+        if (getOwnBuilding() != null)
+        {
+            final int numOfBeesInHive = getOwnBuilding(BuildingBeekeeper.class)
+                                          .getHives()
+                                          .stream()
+                                          .map(world::getTileEntity)
+                                          .filter(Objects::nonNull)
+                                          .map(BeehiveTileEntity.class::cast)
+                                          .mapToInt(BeehiveTileEntity::func_226971_j_)
+                                          .sum();
+            final int numOfAnimals = allAnimals.size();
+            final int maxAnimals = getOwnBuilding().getBuildingLevel() * BEES_PER_LEVEL;
+
+            return (numOfAnimals + numOfBeesInHive) > maxAnimals;
+        }
+        return true;
+    }
+
+    /**
+     * Prepares the beekeeper for herding
+     *
+     * @return The next {@link IAIState}.
+     */
+    private IAIState prepareForHerding()
+    {
+        setDelay(DECIDING_DELAY);
+        if (getOwnBuilding(BuildingBeekeeper.class).shouldHarvestHoneycombs())
+        {
+            if (checkForToolOrWeapon(ToolType.SHEARS))
+            {
+                return getState();
+            }
+        }
+
+        checkIfRequestForItemExistOrCreateAsynch(new ItemStack(Items.GLASS_BOTTLE));
```

Comment:
Possibly, when there are a bunch of glass bottles and flowers in the inventory they shouldn't request

Reply:
I think checkIfRequestForItemExistOrCreateAsynch does already check the inventory

Reply:
> Check if a stack has been requested already or is in the inventory. If not in the inventory and not requested already, create request
from checkIfRequestForItemExistOrCreateAsynch jdoc

Reply:
ah okay